### PR TITLE
`property` of inputs and outputs can be a path

### LINKED ISSUE #25: Property of inputs and outputs should be a path
The `property` name field of callback inputs and outputs should be normalized to a property path. 

```TypeScript
export interface InputOutput {
  kind: InputOutputKind;
  id?: string;
  property: string[];  // A path now!
}
```

Hence, when the UI receives a name from the server, it should be normalized using `input = {...input, property: input.property.split(".")}`.

After the change, we should only use `property` using `get`/`set` as provided by https://github.com/mariocasciaro/object-path.

### PR DESCRIPTION
**Pull request by forman**
Closes #25

### CODE REVIEW

File: dashi/src/lib/utils/objPath.test.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+import { equalObjPaths, getValue, setValue, toObjPath } from "./objPath";
+
+describe("Test that getValue()", () => {
+  it("works on 0th level", () => {
+    const obj = { a: 1, b: "x" };
+    expect(getValue(obj, [])).toEqual({ a: 1, b: "x" });
+  });
+
+  it("works on 1st level", () => {
```
Comment (b-yogesh) [inline dashi/src/lib/utils/objPath.test.ts:11]: We are only testing until the 1st level as described in issue https://github.com/bcdev/dashi/issues/24 ?

File: dashi/src/lib/types/model/callback.ts [hunk 2]
```diff
@@ -62,7 +62,7 @@ export interface CallbackRef {
 export interface InputRef {
   /**
    * The index of the input of a callback.
-   * Used to store the input that triggered the callback.
+   * Records the input that triggered the callback.
```
Comment (b-yogesh) [inline dashi/src/lib/types/model/callback.ts:65]:
```suggestion
   * Records the index of the input that triggered the callback.
```

File: dashi/src/lib/api/types.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+export interface ApiOptions {
```
Comment (b-yogesh) [inline dashi/src/lib/api/types.ts:1]: Can we move these types to the types folder?

**Review by b-yogesh — APPROVED** [review]
Looks good! I have added some questions and some suggestions which can be applied before merging.

File: dashi/src/lib/utils/mapObject.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export function mapObject<V1 = unknown, V2 = unknown>(
+  obj: Record<string, V1>,
+  transformValue: (value: V1, key: string | number) => V2,
+): Record<string, V2> {
+  const obj2: { [k: string]: V2 } = {};
+  Object.getOwnPropertyNames(obj).forEach((key) => {
+    obj2[key] = transformValue(obj[key], key);
+  });
+  return obj2;
```
[the remark this answers is not in the document]
Comment (forman) [reply dashi/src/lib/utils/mapObject.ts:9]: No, but it is there to have a symetry with in-built `Array.prototype.map` function

File: dashi/src/lib/api/types.ts (hunk 3 above)
Comment (forman) [reply dashi/src/lib/api/types.ts:1]: Ok!

**Review by b-yogesh — APPROVED** [review]

**Pull request merged**